tests/e2e/flowtest/test_base_operation_node_drain.py

```diff
@@ -0,0 +1,122 @@
+    """
+    Tests Story/Flow based test scenarios
+
+    """
+    @vsphere_platform_required
+    @aws_platform_required
+    @pytest.mark.polarion_id("OCS-2188")
+    def test_base_operation_node_drain(
+        self, node_drain_teardown, node_restart_teardown, pgsql_factory_fixture,
+        project_factory, multi_pvc_factory, mcg_obj, bucket_factory
+    ):
+        """
+        Test covers following flow operations while running workloads in the background:
+        1. Node drain
+        2. Add capacity
+        3. Node reboot
+        4. Node n/w failure
+
+        """
+        logger.info("Starting IO operations in Background")
+        project = project_factory()
+        bg_handler = flowtest_helpers.BackgroundOps()
+        executor_run_bg_ios_ops = ThreadPoolExecutor(max_workers=3)
+
+        obc_ios = executor_run_bg_ios_ops.submit(
+            bg_handler.handler, obc_put_obj_create_delete, mcg_obj, bucket_factory, iterations=30
+        )
+        logging.info("Started object IOs in background")
+
+        pvc_create_delete = executor_run_bg_ios_ops.submit(
+            bg_handler.handler,
+            create_pvc_delete,
+            multi_pvc_factory,
+            project,
+            iterations=70
+        )
+        logging.info("Started pvc create and delete in background")
+
+        pgsql_workload = executor_run_bg_ios_ops.submit(
+            bg_handler.handler,
+            pgsql_factory_fixture,
+            replicas=1, clients=1, transactions=100,
+            timeout=100, iterations=1
+        )
+        logging.info("Started pgsql workload in background")
+
+        flow_ops = FlowOperations()
+        logger.info("Starting operation 1: Node Drain")
+        node_name, nodes = flow_ops.node_operations_entry_criteria(
+            node_type='worker', number_of_nodes=1, operation_name="Node Drain"
+        )
+        # Node maintenance - to gracefully terminate all pods on the node
+        node.drain_nodes([node_name[0].name])
+        # Make the node schedulable again
+        node.schedule_nodes([node_name[0].name])
+        logger.info("Verifying exit criteria for operation 1: Node Drain")
+        flow_ops.validate_cluster(node_status=True, operation_name="Node Drain")
+
+        logger.info("Starting operation 2: Add Capacity")
+        osd_pods_before, restart_count_before = flow_ops.add_capacity_entry_criteria()
+        # Add capacity
+        osd_size = storage_cluster.get_osd_size()
+        result = storage_cluster.add_capacity(osd_size)
+        pod = OCP(
+            kind=constants.POD, namespace=config.ENV_DATA['cluster_namespace']
+        )
+        pod.wait_for_resource(
+            timeout=300,
+            condition=constants.STATUS_RUNNING,
+            selector='app=rook-ceph-osd',
+            resource_count=result * 3
+        )
+        logger.info("Verifying exit criteria for operation 2: Add Capacity")
+        flow_ops.add_capacity_exit_criteria(restart_count_before, osd_pods_before)
```

Comment:
We should also validate the cluster state after adding a capacity.
May be we can add this check in add_capcity exit criteria?

Comment same for other ops also, basically we should check the system is healthy for every operation.

Reply:
It is validated in add capacity exit as well, checks are validated inside the add capacity exit criteria method.